devicetypes/sensative/sensative-strips-drip-700.src/sensative-strips-drip-700.groovy

```diff
@@ -0,0 +1,559 @@
+@Field static int tempSensorType = 1
+@Field static int leakageCalibrationParamNum = 23
+@Field static int heatAlarm = 4
+@Field static int heatAlarmHigh = 2
+@Field static int heatAlarmLow = 6
+@Field static int waterAlarm = 5
+@Field static int waterAlarmWet = 2
+@Field static int homeSecurity = 7
+@Field static int homeSecurityTamper = 11
+
+metadata {
+	definition (
+		name: "Sensative Strips Drip 700",
+		namespace: "Sensative",
+		author: "Kevin LaFramboise",
+		ocfDeviceType:"x.com.st.d.sensor.moisture",
+		vid: "480ed59e-91d4-3cfc-a077-b06151590ef0",
+		mnmn: "SmartThingsCommunity"
+	) {
+		capability "Sensor"
+		capability "Water Sensor"
+		capability "Temperature Measurement"
+		capability "Tamper Alert"
+		capability "Battery"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Health Check"
+		capability "platemusic11009.firmware"
+		capability "platemusic11009.temperatureAlarm"
+
+		attribute "lastCheckIn", "string"
+
+		fingerprint mfr:"019A", prod:"0004", model:"000B", deviceJoinName: "Strips Drip 700"
+	}
+
+	simulator { }
+
+	preferences {
+		configParams.each { param ->
+			if (param.options) {
+				input "configParam${param.num}", "enum",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					options: param.options
+			} else if (param.range) {
+				input "configParam${param.num}", "number",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+
+		input "debugLogging", "enum",
+			title: "Logging:",
+			required: false,
+			defaultValue: 1,
+			options: [0:"Disabled", 1:"Enabled [DEFAULT]"]
+	}
+}
+
+
+def installed() {
+	logDebug "installed()..."
+	state.pendingRefresh = true
+	initialize()
+}
+
+def updated() {
+	if (!isDuplicateCommand(state.lastUpdated, 1000)) {
+		state.lastUpdated = new Date().time
+
+		logDebug "updated()..."
+		initialize()
+
+		if (!getSettingValue(leakageCalibrationParamNum) && (state.leakageCalibrated != null)) {
+			// reset flag so that it performs calibration the next time it's set to true.
+			logDebug "Resetting leakage/moisture sensor calibration setting..."
+			state.leakageCalibrated = null
+		}
+
+		if (pendingChanges) {
+			logForceWakeupMessage("The configuration changes will be sent to the device the next time it wakes up.")
+		}
+	}
+}
+
+void initialize() {
+	state.debugLoggingEnabled = (safeToInt(settings?.debugOutput, 1) != 0)
+
+	if (!device.currentValue("tamper")) {
+		sendEventIfNew("tamper", "clear")
+	}
+
+	if (!device.currentValue("water")) {
+		sendEventIfNew("water", "dry")
+	}
+
+	if (!device.currentValue("temperatureAlarm")) {
+		sendEventIfNew("temperatureAlarm", "normal")
+	}
+
+	if (!device.currentValue("checkInterval")) {
+		sendEvent(name: "checkInterval", value: ((wakeUpIntervalSeconds * 2) + 300), displayed: falsle, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	}
+}
+
```

Comment:
Please keep one line break between methods (applies to both DTHs from this PR).

Reply:
Resolved.